### PR TITLE
feat: question-answering with large documents using langchain 

### PR DESCRIPTION
**Pull request by inardini**
This PR is about a notebook that shows how to build a question-answering (Q&A) system using LangChain with Vertex AI PaLM API to extract information from large documents.

### CODE REVIEW

File: language/examples/oss-samples/langchain/question_answering_with_large_documents_langchain.ipynb [hunk 1]
```diff
@@ -0,0 +1,1131 @@
+{
```
Comment (polong-lin) [inline language/examples/oss-samples/langchain/question_answering_with_large_documents_langchain.ipynb:1]: <p>2022 -&gt; 2023</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/41/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (polong-lin) [inline language/examples/oss-samples/langchain/question_answering_with_large_documents_langchain.ipynb:1]: <p>Update links</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/41/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (polong-lin) [inline language/examples/oss-samples/langchain/question_answering_with_large_documents_langchain.ipynb:1]: <p>Remove</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/41/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (polong-lin) [inline language/examples/oss-samples/langchain/question_answering_with_large_documents_langchain.ipynb:1]: <p>Check links</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/41/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (polong-lin) [inline language/examples/oss-samples/langchain/question_answering_with_large_documents_langchain.ipynb:1]: <p>Comment out</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/41/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (polong-lin) [inline language/examples/oss-samples/langchain/question_answering_with_large_documents_langchain.ipynb:1]: <p>Header: "Q&amp;A without similarity search"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/41/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='27'/>

Comment (polong-lin) [inline language/examples/oss-samples/langchain/question_answering_with_large_documents_langchain.ipynb:1]: <p>Perhaps add comments or markdown</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/41/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='46'/>

Comment (polong-lin) [inline language/examples/oss-samples/langchain/question_answering_with_large_documents_langchain.ipynb:1]: <p>Header: "Q&amp;A with similarity search"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/41/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='62'/>

Comment (polong-lin) [inline language/examples/oss-samples/langchain/question_answering_with_large_documents_langchain.ipynb:1]: <p>Should perhaps show more examples of Q&amp;A working in action with different questions</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/41/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='72'/>

Comment (polong-lin) [inline language/examples/oss-samples/langchain/question_answering_with_large_documents_langchain.ipynb:1]: <p>May want to consider mentioning Matching Engine as a way for users to go beyond in-memory vector dbs</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/41/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='73'/>

Comment (inardini) [reply language/examples/oss-samples/langchain/question_answering_with_large_documents_langchain.ipynb:1]: <p>done</p>

Comment (inardini) [reply language/examples/oss-samples/langchain/question_answering_with_large_documents_langchain.ipynb:1]: <p>done</p>

Comment (inardini) [reply language/examples/oss-samples/langchain/question_answering_with_large_documents_langchain.ipynb:1]: <p>done</p>

Comment (inardini) [reply language/examples/oss-samples/langchain/question_answering_with_large_documents_langchain.ipynb:1]: <p>done</p>

Comment (inardini) [reply language/examples/oss-samples/langchain/question_answering_with_large_documents_langchain.ipynb:1]: <p>done</p>

Comment (inardini) [reply language/examples/oss-samples/langchain/question_answering_with_large_documents_langchain.ipynb:1]: <p>done</p>

Comment (inardini) [reply language/examples/oss-samples/langchain/question_answering_with_large_documents_langchain.ipynb:1]: <p>I was thinking to be consistent and comparing outputs of all methods with respect the same question</p>

Comment (inardini) [reply language/examples/oss-samples/langchain/question_answering_with_large_documents_langchain.ipynb:1]: <p>done</p>

**Comment by polong-lin** [comment]
LGTM

**Pull request merged**